libs/common-system/src/main/java/org/codice/ddf/configuration/SystemBaseUrl.java

```diff
@@ -0,0 +1,217 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.configuration;
+
+/**
+ * An immutable utility class for getting system url information. Information is pulled from
+ * the system properties.
+ */
+public final class SystemBaseUrl {
+
+    public static final String HTTP_PORT = "org.codice.ddf.system.httpPort";
+
+    public static final String HTTPS_PORT = "org.codice.ddf.system.httpsPort";
+
+    public static final String HOST = "org.codice.ddf.system.hostname";
+
+    public static final String PROTOCOL = "org.codice.ddf.system.protocol";
+
+    public static final String EXT_HTTP_PORT = "org.codice.ddf.system.externalHttpPort";
```

Comment:
Phil and I talked this over and we can't think of any scenario where we wouldn't always just use the "external" host and port and we'd never use the internal one. So lets just get rid of all the "EXT" stuff and stick with the others. It doesn't make any sense for us to be trying to drop all this external stuff in for clustering, when that use case doesn't even exist yet and our assumptions will probably be wrong anyways. We can punt on that until we actually try to configure DDF that way.